Course 13 - GitHub!/Course13Notes.js

```diff
@@ -1,52 +1,97 @@
+    -create a folder with your name on it and add an index.js file that logs something to the console
 
-git branch - shows the available brnahces and highlights to branch you're on
+8. git status once you've saved your changes
+    -this will show you any files that haven't been added or committed yet
 
-git pull - updates your local directory with any new changes????
+9. git add .
+    -this adds all your save changes so you can commit them
 
-git add - adds your changes to something?
-git add . - adds all changes
+10. git commit -m "<brief message here>"
+    -this commits your changes to your local branch
 
-git commit - adds your added changes to something else lol? local branch?
-git commit -m "Message here" - commits changes with a note about the changes
+11. git push
+    -pushes your changes to the remote branch 
+    -can also use git push origin <remote branch name>
 
-git push - pushes commited changes to remote branch
-git push -u - maybe??? why???
+12. You should see your branch in the github now
 
+13. git checkout someone else's branch
+    -doesn't work, need to git pull
 
+14. git pull
+    -now I can checkout a new branch
 
+15. You can create a PR by clicking compare and create PR
+
+16. Go through pull request review process
+
+17. Make sure to always git pull
 */
 
 /*
+Side note: let's talk about stashing changes
+    If you need to switch branches because a more urgent ticket came through, but you're not ready to commit your changes on your current branch.
+        git add .
+            must add your changes to be able to stash them
+        git stash save "<message>"
+            notice reverted changes in text editor
+        git stash list
+            the stash is named. if it's the fist one, it looks like this: stash@{0}
+        git stash apply <stash name>
+            see the changes brought back in text editor
+            apply does not not remove the stash though
+        git stash list
+            see it's still there
+        git stash pop
+            applies changes from first stash in list and removes stash from the list
+        git stash drop stash@{0}
+            removes a single stash
+        git stash clear
+            removes all stashes
+*/
 
-Creating a new repository and connecting it to github
-1. Navigate to a folder you want that you want to make into a github repo
-2. git init - to initialize the git process and create a brand new repo
-3. git add . - to add the files to a staging area WHAT GOOD IS THIS FOR???
-4. git commit -m "<notes>" - to commit the changes to SOMET?HING!!
-5. Log in to your github account and click on new repository and create repository
-6. You should see some instructions there, which we will follow
-7. git remote add origin https://github.com/username/new_repo - WHAT DOES THIS DO SPECIFICALLY?
-8. git push -u origin master - this pushes your local repo to your github account
-9. Check your github account and see that your work has been pushed to github
```

Comment:
Leave this in

Reply:
@znegomir I can add this back into Course 13 if you think it's useful.